apisix/plugins/cors.lua

```diff
@@ -297,13 +297,13 @@ end
 
 function _M.header_filter(conf, ctx)
     local req_origin =  ctx.original_request_origin
-    -- Try allow_origins first, if mismatched, try allow_origins_by_regex.
+    -- If allow_origins_by_regex is not nil, should be matched to it only
     local allow_origins
-    allow_origins = process_with_allow_origins(conf.allow_origins, ctx, req_origin)
-    if not match_origins(req_origin, allow_origins) then
-        allow_origins = process_with_allow_origins_by_regex(conf, ctx, req_origin)
+    allow_origins = process_with_allow_origins_by_regex(conf, ctx, req_origin)
```

Comment:
Look like we can skip this function call by checking the `conf.allow_origins_by_regex == nil` first. For example,

```
if conf.allow_origins_by_regex ~= nil then
```

Reply:
![image](https://user-images.githubusercontent.com/110583072/225187467-e4099caa-7c13-4671-b8ff-f57514ac0447.png)

This function named  process_with_allow_origins_by_regex has checked  whether allow_origins_by_regex is  nil. 
Does it need to be checked again before the method invocation?

Reply:
Yes. We can move it out, and avoid the repeated `and conf.allow_origins_by_regex == nil` check below.

Reply:
Thanks  and I has changed it.